### PR TITLE
create models and relationship between them

### PR DESCRIPTION
**Pull request by vjnhat**

### CODE REVIEW

**Comment by vanlinha3k35pbc** [comment]
OK

File: app/models/activity.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Activity < ActiveRecord::Base
+  belongs_to :user
+  has_many :like_unlikes, dependent: :destroy
+  
+  enum action: [:read, :mark, :like, :unlike, :write, :comment, :follow, :unfollow, :request]
```
Comment (dieunb) [inline app/models/activity.rb]: Những action này em nên cho vào Setting

File: app/models/activity.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Activity < ActiveRecord::Base
+  belongs_to :user
+  has_many :like_unlikes, dependent: :destroy
+
+  enum action: Settings.action
```
Comment (dieunb) [inline app/models/activity.rb]: `actions` not `action`

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  has_many :passive_relationships, class_name:  "Relationship",
+                                   foreign_key: "followed_id",
+                                   dependent: :destroy
+
+  has_many :following, through: :active_relationships,
+                       source: :followed
+  has_many :followers, through: :passive_relationships,
+                       source: :follower
+
+  enum admin: [:normal, :admin]
```
Comment (dieunb) [inline app/models/user.rb]: `role` instead of `admin`

File: app/models/activity.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Activity < ActiveRecord::Base
+  belongs_to :user
+  has_many :like_unlikes, dependent: :destroy
```
Comment (dieunb) [inline app/models/activity.rb]: simple is `likes`

**Pull request closed without merging**